File: src/main/java/net/dankrushen/danknn/danklayers/DankLayer.java
package net.dankrushen.danknn.danklayers;

import net.dankrushen.danknn.DankNeuron;

public abstract class DankLayer implements IDankLayer {
    private DankNeuron[] neurons;

    public DankLayer(int numNeurons) {
        neurons = new DankNeuron[numNeurons];

        for (int i = 0; i < numNeurons; i++) {
            neurons[i] = new DankNeuron(this);
        }
    }

    @Override
    public DankNeuron[] getNeurons() {
        return neurons;
    }

    @Override
    public void setBias(double bias) {
        for (DankNeuron neuron : getNeurons()) {
            neuron.setBias(bias);
        }
    }
}

File: src/main/java/net/dankrushen/danknn/danklayers/DankLayer.kt
package net.dankrushen.danknn.danklayers

import net.dankrushen.danknn.DankNeuron

abstract class DankLayer(numNeurons: Int) : IDankLayer {
    final override var neurons: Array<DankNeuron> = arrayOf()

    init {
        val neuronList = mutableListOf<DankNeuron>()

        for (i in 0 until numNeurons) {
            neuronList.add(DankNeuron(this))
        }

        neurons = neuronList.toTypedArray()
    }

    override fun setBias(bias: Double) {
        for (neuron in neurons) {
            neuron.bias = bias
        }
    }
}
